eth_utils/address.py

```diff
@@ -35,25 +33,23 @@ def is_hex_address(value):
         return False
 
 
-@coerce_args_to_bytes
 def is_binary_address(value):
     """
-    Checks if the given string is an address in raw bytes form.
+    Checks if a given string is an address in raw bytes form.
     """
-    if not is_string(value):
+    if not is_bytes(value):
         return False
     elif len(value) != 20:
         return False
     else:
         return True
 
 
-@coerce_args_to_text
 def is_32byte_address(value):
     """
-    Checks if the given string is an address in hexidecimal encoded form padded to 32 bytes.
+    Checks if a given string is an address in hexidecimal encoded form padded to 32 bytes.
     """
-    if not is_string(value):
+    if not is_text(value):
         return False
 
     if len(value) == 32:
```

Comment:
I'm realizing this line is in contradiction to what the docstring for this function says.  @carver got any thoughts on this?  Should we support binary encoded addresses?  Should we split this function into two?

- `is_32byte_binary_address`
- `is_32byte_hex_address`

Not sure what the right thing to do here is, and honestly, I think **most** of the use cases for this function are theoretical.

Reply:
Thanks for the help @carver and @pipermerriam ! Waiting on confirmation whether/not to drop 32byte support from Piper, and I'll push the updated version.

Reply:
@pipermerriam thoughts on dropping support for 32-byte addresses?